### DISCUSSION TITLE
Split Matrix block button

### DISCUSSION THREAD

**Discussion by proimage** [craftcms/redactor#396] [Ideas]
I'd love to have a button in Matrix block longform text fields—both WYSIWYG and, if possible, textarea—that will duplicate the Matrix block while splitting the textual content before and after the cursor position between the original and new blocks. This would allow easy insertion of images, pull quotes, and the like into the middle of existing text blocks.

If you're feeling really adventurous, then perhaps instead of merely splitting a block, you could have the button directly insert a block of the user's choice into the specified position (so if you choose to insert an `image` block into the middle of a `text` block, you'd end up with: `text`-`image`-`text`).

**Comment by Mosnar** [comment]
Just stumbled across this ancient issue and I doubt this will help anyone, but I happen to have written a Craft 3 plugin to do just this:
https://plugins.craftcms.com/redactor-split
